enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,442 @@
+     `ovnkube-node` can be started.
+
+     The script will also add the node annotation `k8s.ovn.org/node-subnets`
+     according to the HostSubnet CR of this node.
+     
+   CNO wait until OVN-K is deployed in the cluster Set the
+   `LiveMigrationTargetCNIReady` condition to `TRUE` in the status of the
+   network.config CR.
+
+6. CNO updates the `.status.migration.networkType` of the network.config CR.
+   This change will trigger MCO to apply a new MachineConfig to each MCP:
+
+   - MCO will render a new MachineConfig for each MCP.
+   - MCO will cordon, drain and reboot the node.
+   - When the node is up, the `br-ex` bridge will be created. It means the node
+     is ready for OVN Kubernetes. So the local zone OVN Kubernetes components
+     can be started. The ovnkube-node container wrapper script will also add the
+     node annotation `k8s.ovn.org/node-subnets`, according to the HostSubnet CR
+     of this node. On the local node, OVN Kubernetes will add all SDN nodes as
+     hybrid overlay nodes to the local zone and all OVN nodes as OVN-IC remote
+     zones. On a remote OVN node, OVN Kubernetes will remove a hybrid overlay
+     node and add an OVN-IC remote zone. Pods will be recreated on the node
+     using ovn-kubernetes as the default CNI plugin.
+   - MCO will uncordon the node. New pods can be scheduled to this node.
+
+   The above process will be repeated for each node until all the nodes have
+   been applied to the new MachineConfig and converted to OVN Kubernetes.
+
+7. CNO watches the MCPs, wait until the new MCs is applied to all the nodes. Set
+   the `LiveMigrationTargetMachineConfigApplied` condition to `TRUE` in the
+   status of the network.config CR.
+
+8. CNO removes the `spec.migration` field of network.operator CR. It will
+   trigger the CNO to:
+
+   - delete the openshift-sdn DaemonSets and the related resources
+     (CustomResources, ConfigMaps, etc.).
+   - redeploy ovn-kubernetes in "normal" mode (no migration mode script, hybrid
+     overlay disabled, etc.).
+   - remove the migration-related labels and node annotations from the nodes.
+
+9. CNO wait until OVN-K is redeployed Set the `LiveMigrationOriginalCNIPurged`
+   condition to `TRUE` in the status of the network.config CR. Then set the
+   `LiveMigrationProgressing` condition to `FALSE` with the reason
+   `LiveMigrationCompleted` in the status of the network.config CR. This
+   indicates that the migration is completed.
+
+### API
+
+For `Network.config.openshift.io` CRD, we will use the CR annotation
+`network.openshift.io/live-migration:` to gate the modification to
+`spec.networkType` field. If the annotation absents (default), CNO will block
+the change to take effect. If it exists, CNO takes it as the trigger of starting
+SDN live migration.
+
+The following conditions will be added to the `.status` of the
+`Network.config.openshift.io` CR:
+
+- LiveMigrationProgressing
```

Comment:
Is it ok that `LiveMigrationProgressing` is never set to false if it is stuck or failed for whatever reason? Should this read differently, like `LiveMigrationInProgress`.

Reply:
Yes, if `LiveMigrationProgressing` got stuck in `FALSE`. It means the is an error in the live migration. It's an indicator to the user that he needs to fix the issue first.

Reply:
Perhaps worth mentioning somewhere as well. I read that the only reason for this to be set to false is if non-migratable features are in use.

Also, if there is no mention of any specific implementation needed to support the remediation of such cases, I guess that means that the procedure is supposed to automatically resume progressing once the user has fixed the issue or that otherwise it needs to be handled on a case by case basis through support? I think it would be useful to have a mention about this as well.

Reply:
I added more details. PTAL.